packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -1144,15 +1398,16 @@ namespace Ifpack2 {
             Kokkos::parallel_for(Kokkos::ThreadVectorRange(member, vector_loop_size),[&](const int &v) {
                 Kokkos::parallel_for(Kokkos::TeamThreadRange(member,icount),[&](const local_ordinal_type &ii) {
                     const local_ordinal_type i = ibeg + ii*3;
-                    for (local_ordinal_type j=0;j<blocksize;++j)
-                      values(i,j,j,v) = 1;
+                    for (local_ordinal_type j0=0;j0<blocksize;++j0) 
+                      for (local_ordinal_type j1=0;j1<blocksize;++j1) 
+                        values(i,j0,j1,v) = (j0 == j1 ? 1 : 0);
```

Comment:
This will likely fail to compile in the complex arithmetic case, since there's no assignment defined from integers to `complex<double>`.  The easiest fix is just to use `1.0` and `0.0` instead of 1 and 0.

Reply:
This so far have been passed for complex tests but will change it using one and zero. done.

Reply:
Thanks!